internal/interpreter/api.rs

```diff
@@ -1017,6 +1017,25 @@ impl ComponentInstance {
     pub fn highlight(&self, path: PathBuf, offset: u32) {
         crate::highlight::highlight(&self.inner, path, offset);
     }
+
+    /// Request information on clicked object
+    ///
+    /// WARNING: this is not part of the public API
+    #[cfg(feature = "highlight")]
+    pub fn design_mode(&self, active: bool) {
+        crate::highlight::design_mode(&self.inner, active);
+    }
+
+    /// Register callback to handle current item information
+    ///
+    /// WARNING: this is not part of the public API
+    #[cfg(feature = "highlight")]
+    pub fn set_current_element_information_callback(
```

Comment:
We need a better name for the callback. Maybe it should describe when the callback is fired:
 `set_current_element_changed_callback` ?
 `set_element_selected_callback` ?
`on_element_selected` ?